executor/analyze.go

```diff
@@ -262,6 +270,30 @@ func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
 	return nil
 }
 
+func (e *AnalyzeExec) recordHistoricalStats(tableID int64) error {
+	statsHandle := domain.GetDomain(e.ctx).StatsHandle()
+	historicalStatsEnabled, err := statsHandle.CheckHistoricalStatsEnable()
+	if err != nil {
+		return errors.New("check tidb_enable_historical_stats failed")
```

Comment:
The underlying error message should be contained in the returned error:
```suggestion
		return errors.Errorf("check tidb_enable_historical_stats failed: %v", err)
```